Formula/s/screenpipe.rb

```diff
@@ -0,0 +1,38 @@
+class Screenpipe < Formula
+  desc "Library to build personalized AI powered by what you've seen, said, or heard"
+  homepage "https://github.com/mediar-ai/screenpipe"
+  url "https://github.com/mediar-ai/screenpipe/archive/refs/tags/v0.1.98.tar.gz"
+  sha256 "cb3c8039ecb60d35bacd2b9673db112f907b4a1d3d7c32f49a5e77c0274268ad"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "rust" => :build
+  depends_on "ffmpeg"
+  depends_on :macos
+
+  on_macos do
+    def install
+      ENV["PKG_CONFIG_PATH"] = "/usr/local/opt/ffmpeg/lib/pkgconfig:#{ENV["PKG_CONFIG_PATH"]}" if Hardware::CPU.intel?
+      ENV["PKG_CONFIG_ALLOW_CROSS"] = "1" if Hardware::CPU.intel?
+
+      target_arch = Hardware::CPU.arm? ? "aarch64-apple-darwin" : "x86_64-apple-darwin"
+      system "cargo", "build", "--release", "--features", "metal,pipes", "--target", target_arch
+      bin.install "target/#{target_arch}/release/screenpipe"
+      lib.install "target/#{target_arch}/release/libscreenpipe.dylib"
+      lib.install "target/#{target_arch}/release/libscreenpipe_#{Hardware::CPU.arch}.dylib"
+    end
+
+    def caveats
+      <<~EOS
+        Bun is not available on Homebrew
+        To install Bun, please visit the official website:
+        https://bun.sh
+        Follow the instructions there to install Bun on your system.
+      EOS
+    end
+  end
+
+  test do
+    system "#{bin}/screenpipe", "-h"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula